src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -608,6 +566,30 @@ Lit CoreSMTSolver::pickBranchLit()
 
 }
 
+/**
+ * Compute the Glue value, as defined in Audemard & Simon:
+ * Predicting Learnt Clauses Quality in Modern SAT Solvers.
+ * IJCAI 2009.
+ *
+ * @param vector of literals each having a level in vardata
+ * @return min (4, |{level(var(lit))}| \mid lit \in ps)
+ */
+uint32_t CoreSMTSolver::computeGlue(vec<Lit> const & ps) {
+    levelsInClause.reset();
+    uint32_t numLevels = 0;
+    for (Lit lit: ps) {
+        int level = vardata[var(lit)].level;
+        if (level != 0 and not levelsInClause.contains(level)) {
+            levelsInClause.insert(level);
+            ++ numLevels;
+            if (numLevels >= 4) {
+                break;
```

Comment:
I think you can go to 7 here, so maybe you want to up this `4` to `7`? More resolution does seem to help. 4 seems very low, even 7 may be too low IMHO.